closed/autoconf/custom-hook.m4

```diff
@@ -53,22 +53,22 @@ AC_DEFUN_ONCE([CUSTOM_EARLY_HOOK],
 AC_DEFUN([OPENJ9_CONFIGURE_CMAKE],
 [
   AC_ARG_WITH(cmake, [AS_HELP_STRING([--with-cmake], [enable building openJ9 with CMake])],
+    [],
     [
-      if test "x$with_cmake" == xyes -o "x$with_cmake" == x ; then
-        with_cmake=cmake
-      fi
-      if test "x$with_cmake" != xno ; then
-        if AS_EXECUTABLE_P(["$with_cmake"]) ; then
-          CMAKE="$with_cmake"
-        else
-          BASIC_REQUIRE_PROGS([CMAKE], [$with_cmake])
-        fi
+      # Determine a default if neither --with-cmake nor --without-cmake was given
+      if test "x$OPENJDK_BUILD_OS" = xmacosx ; then
         with_cmake=yes
+      else
+        with_cmake=no
       fi
-    ],
-    [with_cmake=no])
-  if test "$with_cmake" == yes ; then
+    ])
+
+  if test "$with_cmake" != no ; then
     OPENJ9_ENABLE_CMAKE=true
+    BASIC_PATH_PROGS(CMAKE, cmake)
```

Comment:
This used to support `--with-cmake=alternate-cmake-version`: please restore that capability.

Reply:
This can still be achived by setting the `CMAKE` environment variable/option on the command line. This is more in line with how you would specify alternate versions of other tools. eg setting the `CC` variable

Reply:
If the configure option were `--enable-cmake` I would agree with you, but the convention is that `--with-XXX` options accept values (see also `--with-openj9-cc` and  `--with-openj9-cxx`, etc.). Please don't break existing behavior just to change the default for one platform.